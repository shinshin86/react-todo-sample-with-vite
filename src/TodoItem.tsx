export type Todo = {
  readonly id: number;
  text: string;
  checked: boolean;
};

type Props = {
  todo: Todo;
  onCheckTodo: Function;
};

const TodoItem: React.FC<Props> = ({ todo, onCheckTodo }) => (
  <div>
    <input
      type="checkbox"
      id={todo.id.toString()}
      defaultChecked={todo.checked}
      onChange={() => onCheckTodo(todo.id)}
    />
    <label htmlFor={todo.id.toString()}>{todo.text}</label>
  </div>
);

export default TodoItem;
