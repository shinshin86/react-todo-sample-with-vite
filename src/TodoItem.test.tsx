import { describe, expect, test } from "vitest";
import "@testing-library/jest-dom";
import TodoItem, { Todo } from "./TodoItem";

import { render, screen } from "@testing-library/react";

describe("Simple working test", () => {
  const testText = "testtext";
  const todo: Todo = {
    id: 1,
    text: testText,
    checked: false,
  };
  const emptyOnCheckTodo = () => console.log("emptyOnCheckTodo");

  test("Should render correctly", () => {
    const { container } = render(
      <TodoItem todo={todo} onCheckTodo={emptyOnCheckTodo} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("Should display text", () => {
    render(
      <TodoItem todo={todo} onCheckTodo={emptyOnCheckTodo} />,
    );

    expect(screen.getByText(testText)).toBeInTheDocument();
  });

  test("Should display unchecked checkbox", () => {
    const { container } = render(
      <TodoItem todo={todo} onCheckTodo={emptyOnCheckTodo} />,
    );

    expect(container.querySelector("input")).not.toBeChecked();
  });
});
