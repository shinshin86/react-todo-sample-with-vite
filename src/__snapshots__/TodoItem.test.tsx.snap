// Vitest Snapshot v1

exports[`Simple working test > Should render correctly 1`] = `
<div>
  <input
    id="1"
    type="checkbox"
  />
  <label
    for="1"
  >
    testtext
  </label>
</div>
`;
